partner_event/models/event_registration.py

```diff
@@ -35,19 +36,17 @@ def create(self, vals):
             attendee_partner = Partner.search([("email", "=ilike", email)], limit=1)
             event = Event.browse(vals["event_id"])
             if attendee_partner:
-                vals["name"] = vals.setdefault("name", attendee_partner.name)
-                vals["phone"] = vals.setdefault("phone", attendee_partner.phone)
```

Comment:
Out of curiosity. Why didn't this work?

Reply:
https://github.com/OCA/event/pull/188#discussion_r484867623